components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/handlers/SCIMClaimOperationEventHandler.java

```diff
@@ -104,4 +119,37 @@ public String getName() {
 
         return "SCIMClaimOperationEventHandler";
     }
+
+    private void handleSCIMExternalClaimAddEvent(Event event) {
+
+        if (!IdentityEventConstants.Event.PRE_ADD_EXTERNAL_CLAIM.equals(event.getEventName())) {
+            return;
+        }
+
+        String claimDialectUri = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.CLAIM_DIALECT_URI);
+        String externalClaimUri = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.EXTERNAL_CLAIM_URI);
+        Set<String> scimClaimDialects = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+                SCIM_CORE_CLAIM_DIALECT,
+                SCIM_USER_CLAIM_DIALECT,
+                SCIM_ENTERPRISE_USER_CLAIM_DIALECT,
+                SCIM_SYSTEM_USER_CLAIM_DIALECT
+        )));
+
+        /*
+         * All spec-defined claims might not be added to dialects as external claims. All supported claims can be found
+         * through the schemas.profile config defined in the identity.xml. The DialectConfigParser is used to read the
+         * final server supported claims.
+         */
+        if (DialectConfigParser.getInstance().getClaimsMap().get(externalClaimUri) != null
+                && DialectConfigParser.getInstance().getClaimsMap().get(externalClaimUri).equals(claimDialectUri)) {
+            return;
+        }
+
```

Comment:
Here we can remove the thread local for the safe side.

Reply:
Better to remove it in the place we use them inside a finally block. Refer https://github.com/wso2/carbon-identity-framework/blob/59edad692c5cce276a5a151f9e9ec046542d1d26/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java#L209

Reply:
If we haven't done it shall we fix it?